parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -227,6 +239,65 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return internalReader.nextKeyValue();
   }
 
+  /**
+   * Currently reading a batch of rows of complex types is not supported
+   *
+   * Reads the next batch of rows. This method is used for reading complex types
+   * or arbitrary objects and calls the converters eventually to materialize the record.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @param clazz the class of the record type that will be filled into the column vector
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  public RowBatch nextBatch(RowBatch previous, Class<T> clazz) throws IOException {
+    throw new UnsupportedOperationException("Reading a batch of rows of complex types is not supported");
+  }
```

Comment:
let's remove this if we don't use it. I don't see how the lower layer would know about T anyway.